nrf_modem: v2.3.0 (#910)
nrfconnect/sdk-nrfxlib: nrf_modem/doc/architecture.rst

Shared memory configuration
***************************

The shared memory between the application and modem cores must reside within the lowest 128 kilobytes of RAM.
The application is responsible for reserving the memory area before passing it onto the library.

> In bootloader mode, the shared memory area must be as large as ``NRF_MODEM_SHMEM_BOOTLOADER_SIZE`` and its base address must be 4-bytes aligned.

Reviewer: ```suggestion
In bootloader mode, the shared memory area must be as large as the value of ``NRF_MODEM_SHMEM_BOOTLOADER_SIZE`` and its base address must be 4-bytes aligned.
```

> The library accepts the shared memory base address and size as parameters to the :c:func:`nrf_modem_bootloader_init` function through the :c:struct:`nrf_modem_bootloader_init_params`.

Reviewer: ```suggestion
The library accepts the shared memory base address and size as parameters to the :c:func:`nrf_modem_bootloader_init` function through the :c:struct:`nrf_modem_bootloader_init_params` structure.
```

> In normal operating mode the shared memory is divided into the following four regions:

Reviewer: ```suggestion
In normal operating mode, the shared memory is divided into the following four regions:
```

* Control
* TX
* RX
* Trace

> The application can adjust the size of these regions based on its requirements, with the exception of the Control region which has a fixed size that must be equal to ``NRF_MODEM_SHMEM_CTRL_SIZE``.

Reviewer: ```suggestion
The application can adjust the size of these regions based on its requirements, with the exception of the Control region which has a fixed size that must be equal to the value of  ``NRF_MODEM_SHMEM_CTRL_SIZE``.
```

The base address of all regions must be 4-bytes aligned.

> The library accepts the layout of these regions as parameters to the :c:func:`nrf_modem_init` function through the :c:struct:`nrf_modem_shmem_cfg` struct in :c:struct:`nrf_modem_init_params`.

Reviewer: ```suggestion
The library accepts the layout of these regions as parameters to the :c:func:`nrf_modem_init` function through the :c:struct:`nrf_modem_shmem_cfg` structure in :c:struct:`nrf_modem_init_params`.
```

* :c:func:`nrf_modem_os_shm_tx_alloc`
* :c:func:`nrf_modem_os_shm_tx_free`

> .. important:
>    It must be possible for the Modem library to allocate up to the number of bytes passed by the TX region size in the :c:struct:`nrf_modem_init_params` struct.
>    If the OS implementation of :c:func:`nrf_modem_os_shm_tx_alloc` require any overhead, additional memory must be set aside by the application.

Reviewer: ```suggestion
.. important::
   It must be possible for the Modem library to allocate up to the number of bytes passed by the TX region size in the :c:struct:`nrf_modem_init_params` structure.
   If the OS implementation of :c:func:`nrf_modem_os_shm_tx_alloc` requires any overhead, additional memory must be set aside by the application.
```
